### DISCUSSION TITLE
`git-machete`

### DISCUSSION THREAD

**Discussion by arxanas** [gitext-rs/git-stack#226] [General]
Saw this project: https://github.com/VirtusLab/git-machete

I don't fully understand it, but I think it occupies the same space as `git-stack`. There might be some interesting ideas around automatically discovering protected/unprotected branches and saving the configuration.

**Comment by epage** [comment]
Thanks for pointing it out.  https://github.com/gitext-rs/git-stack/pull/227 includes updating it to the comparison.

I had seen other tools use files to track branch relationships (can't remember which it was atm).  The main problem I have with them is it adds too much weight to a branch.
